sde_collections/tasks.py

```diff
@@ -141,3 +143,59 @@ def resolve_title_pattern(title_pattern_id):
     TitlePattern = apps.get_model("sde_collections", "TitlePattern")
     title_pattern = TitlePattern.objects.get(id=title_pattern_id)
     title_pattern.apply()
+
+
+@celery_app.task
+def fetch_and_update_full_text(collection_id, server_type):
+    try:
+        collection = Collection.objects.get(id=collection_id)
+    except Collection.DoesNotExist:
+        raise Exception(f"Collection with ID {collection_id} does not exist.")
+
+    server_config = get_server_config(server_type)
+    token = server_config["token"]
+    url = server_config["url"]
+
+    headers = {"Content-Type": "application/json", "Authorization": f"Bearer {token}"}
+
+    payload = json.dumps(
+        {
+            "method": "engine.sql",
+            "sql": f"SELECT url1, text, title FROM sde_index WHERE collection = '/SDE/{collection.config_folder}/'",
+            "pretty": True,
+            "log": False,
+            "output": "json",
+            "resolveIndexList": "false",
+            "engines": "default",
+        }
+    )
+
+    try:
+        response = requests.post(url, headers=headers, data=payload, timeout=10)
+        response.raise_for_status()  # Raise exception for HTTP errors
+    except requests.exceptions.RequestException as e:
+        raise Exception(f"API request failed: {str(e)}")
```

Comment:
this handling code should be part of the api class as well. potentially as a dedicated function that is used with every type of api query, including the existing query function

Reply:
Error handling included as part of the Api class extenstion